compiler/rustc_const_eval/src/interpret/call.rs

```diff
@@ -189,7 +189,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 ty::Ref(_, ty, _) => *ty,
                 ty::RawPtr(ty, _) => *ty,
                 // We only accept `Box` with the default allocator.
-                _ if ty.is_box_global(*self.tcx) => ty.boxed_ty(),
+                _ if ty.is_box_global(*self.tcx) => ty.boxed_ty_unchecked(),
```

Comment:
"unchecked" is usually used for unsafe operations. Please use a less scary name.

Reply:
IME `expect_*` is somewhat common in the compiler. Could use `expect_boxed_ty`

Reply:
Yeah, that or something with `unwrap`.

Reply:
Or you can use `boxed_ty` and `try_boxed_ty`?

Reply:
I think `expect_boxed_ty` and `boxed_ty` are the best. We do that for other things for example like `exect_ty` and `ty` on `GenericArg`.